pkg/controller/gather_commands.go

```diff
@@ -0,0 +1,243 @@
+	updatedCR.Status.StartTime = metav1.Now()
+
+	dataGatherCR, err = insightClient.DataGathers().UpdateStatus(ctx, updatedCR, metav1.UpdateOptions{})
+	if err != nil {
+		klog.Error("failed to update coresponding DataGather custom resource: %v", err)
+		return err
+	}
+
+	// ensure the insight snapshot directory exists
+	if _, err = os.Stat(d.StoragePath); err != nil && os.IsNotExist(err) {
+		if err = os.MkdirAll(d.StoragePath, 0777); err != nil {
+			return fmt.Errorf("can't create --path: %v", err)
+		}
+	}
+
+	// configobserver synthesizes all config into the status reporter controller
+	configObserver := configobserver.New(d.Controller, kubeClient)
+	// anonymizer is responsible for anonymizing sensitive data, it can be configured to disable specific anonymization
+	anonymizer, err := anonymization.NewAnonymizerFromConfig(
+		ctx, gatherKubeConfig, gatherProtoKubeConfig, protoKubeConfig, configObserver, dataGatherCR.Spec.DataPolicy)
+	if err != nil {
+		return err
+	}
+
+	// the recorder stores the collected data and we flush at the end.
+	recdriver := diskrecorder.New(d.StoragePath)
+	rec := recorder.New(recdriver, d.Interval, anonymizer)
+	authorizer := clusterauthorizer.New(configObserver)
+	insightsClient := insightsclient.New(nil, 0, "default", authorizer, gatherKubeConfig)
+
+	gatherers := gather.CreateAllGatherers(
+		gatherKubeConfig, gatherProtoKubeConfig, metricsGatherKubeConfig, alertsGatherKubeConfig, anonymizer,
+		configObserver, insightsClient,
+	)
+	uploader := insightsuploader.New(nil, insightsClient, configObserver, nil, nil, 0)
+
+	allFunctionReports := make(map[string]gather.GathererFunctionReport)
+	for _, gatherer := range gatherers {
+		functionReports, err := gather.CollectAndRecordGatherer(ctx, gatherer, rec, dataGatherCR.Spec.Gatherers) // nolint: govet
+		if err != nil {
+			klog.Errorf("unable to process gatherer %v, error: %v", gatherer.GetName(), err)
+		}
+
+		for i := range functionReports {
+			allFunctionReports[functionReports[i].FuncName] = functionReports[i]
+		}
+	}
+	err = gather.RecordArchiveMetadata(mapToArray(allFunctionReports), rec, anonymizer)
+	if err != nil {
+		klog.Error(err)
+		return err
+	}
+	err = rec.Flush()
+	if err != nil {
+		klog.Error(err)
+		return err
+	}
+	lastArchive, err := recdriver.LastArchive()
+	if err != nil {
+		klog.Error(err)
+		return err
+	}
+	insightsRequestID, err := uploader.Upload(ctx, lastArchive)
+	if err != nil {
+		klog.Error(err)
+		return err
+	}
+	klog.Infof("Insights archive successfully uploaded with InsightsRequestID: %s", insightsRequestID)
+
+	dataGatherCR.Status.FinishTime = metav1.Now()
+	dataGatherCR.Status.State = insightsv1alpha1.Completed
+	dataGatherCR.Status.InsightsRequestID = insightsRequestID
+	dataGatherCR.Status.Conditions = []metav1.Condition{
+		{
+			Type:               "DataUploaded",
+			Status:             metav1.ConditionTrue,
+			Reason:             "AsExpected",
+			LastTransitionTime: metav1.Now(),
+		},
+	}
```

Comment:
This status updating is temporary and will be addressed in https://issues.redhat.com/browse/CCXDEV-10590